linter/src/crow/crow.cpp

```diff
@@ -43,83 +44,44 @@ get_server_thread()
-            // LINT_ABSOLUTE_TIMEOUT_SECONDS seconds (default 130s)
-            std::thread check_pending_thread([algo_id, uid]() {
-                for (int i = 0; i <= LINT_ABSOLUTE_TIMEOUT_SECONDS; i++) {
-                    if (nutc::client::get_algo_status(uid, algo_id)
-                        != nutc::client::LintingResultOption::PENDING) {
-                        log_i(
-                            crow_watchdog,
-                            "Algo id {} for uid {} firebase result detected as set by "
-                            "crow-level watchdog {}s",
-                            algo_id,
-                            uid,
-                            i
-                        );
-                        return;
-                    }
-                    std::this_thread::sleep_for(std::chrono::seconds(1));
-                }
-                // Check status from Firebase
-                nutc::client::LintingResultOption linting_status =
-                    nutc::client::get_algo_status(uid, algo_id);
+            std::optional<std::string> algo_code = nutc::client::get_algo(uid, algo_id);
+            if (!algo_code.has_value()) {
+                nutc::client::set_lint_failure(
+                    uid,
+                    algo_id,
+                    fmt::format(
+                        "[linter] FAILURE - could not find algo {} for id {}\n",
+                        algo_id,
+                        uid
+                    )
+                );
+                crow::json::wvalue response = crow::json::wvalue({
+                    {"linting_status",
+                     static_cast<int>(client::LintingResultOption::UNKNOWN)}
+                });
+
+                res.body = response.dump();
+                res.code = 200;
 
-                // If status is pending, force push a failure
-                switch (linting_status) {
-                    case nutc::client::LintingResultOption::PENDING:
-                        {
-                            // Push failure
-                            std::string error_msg = fmt::format(
-                                "Internal server error. Reach out to "
-                                "nuft@u.northwestern.edu for support",
-                                LINT_ABSOLUTE_TIMEOUT_SECONDS
-                            );
+                return res;
+            }
 
-                            nutc::client::set_lint_failure(uid, algo_id, error_msg);
+            client::LintingResultOption algo_status_code;
+            auto lint_res = spawner_manager.spawn_client(algo_code.value());
+            std::cout << lint_res.message << '\n';
+            if (lint_res.success) {
+                nutc::client::set_lint_success(uid, algo_id, lint_res.message);
+                algo_status_code = client::LintingResultOption::SUCCESS;
+            }
+            else {
+                nutc::client::set_lint_failure(uid, algo_id, lint_res.message);
```

Comment:
these web requests could (and probably should) be done asynchronously.

Reply:
the current design is single threaded and we don't have a reason to change that right now since the linter is not very performance critical